### ISSUE TITLE
Podspec mode doesn't pass through swift version 4.2 properly

### ISSUE THREAD

**Issue by tackgyu** [realm/jazzy#1015]
I tried generating swift docs named `TDS` using jazzy using following command:
`jazzy --podspec LocalPods/TDS/TDS.podspec --swift-version 4.2`

and also tried with config file `.jazzy.yaml` like:

```
module: TDS
swift_version: 4.2
```

using following command:
`jazzy --podspec LocalPods/TDS/TDS.podspec --config LocalPods/TDS/.jazzy.yaml`

but both failed with the following output:
```
Running xcodebuild
Could not parse compiler arguments from `xcodebuild` output.
Please confirm that `xcodebuild` is building a Swift module.
Saved `xcodebuild` log file: /var/folders/0_/wjng6zns5r9btwdgnt7hcxc00000gn/T/xcodebuild-A5CA7AF9-153E-4658-A443-46B8456E5373.log
Failed to generate documentation
bundler: failed to load command: jazzy (/Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bin/jazzy)
RuntimeError: /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/bin/sourcekitten ["doc", "--module-name", "TDS", "--", "-target", "TDS", "SWIFT_VERSION\\=4.0"]

Running xcodebuild

Could not parse compiler arguments from `xcodebuild` output.

Please confirm that `xcodebuild` is building a Swift module.

Saved `xcodebuild` log file: /var/folders/0_/wjng6zns5r9btwdgnt7hcxc00000gn/T/xcodebuild-A5CA7AF9-153E-4658-A443-46B8456E5373.log

Failed to generate documentation

  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/executable.rb:36:in `execute_command'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/sourcekitten.rb:230:in `run_sourcekitten'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/podspec_documenter.rb:32:in `block (3 levels) in sourcekitten_output'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/podspec_documenter.rb:28:in `map'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/podspec_documenter.rb:28:in `block (2 levels) in sourcekitten_output'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/podspec_documenter.rb:23:in `chdir'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/podspec_documenter.rb:23:in `block in sourcekitten_output'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/gems/cocoapods-1.5.3/lib/cocoapods/config.rb:41:in `with_changes'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/podspec_documenter.rb:18:in `sourcekitten_output'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/lib/jazzy/doc_builder.rb:62:in `build'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/bin/jazzy:15:in `<top (required)>'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bin/jazzy:23:in `load'
  /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bin/jazzy:23:in `<top (required)>'
```

It seems to me that the swift version I passed as paramter is somehow ignored while jazzy passing its parameters to sourcekitten, regarding the line in the output says:
`RuntimeError: /Users/tackgyu/Sources/toss-ios/vendor/bundle/ruby/2.4.0/bundler/gems/jazzy-6932e5efe0eb/bin/sourcekitten ["doc", "--module-name", "TDS", "--", "-target", "TDS", "SWIFT_VERSION\\=4.0"]`.

Does jazzy just not support Swift 4.2 yet, or can you at least let me know if there's a workaround for this issue if any?

**Comment by bartekzabicki** [comment]
@tackgyu Did you find any solution for this? I have the same problem

**Comment by tackgyu** [comment]
@bartekzabicki Still nothing. Just have to wait for it to support Swift 4.2

**Comment by erdnussflips** [comment]
Could this PR fix that problem: https://github.com/realm/jazzy/pull/1019 ?

**Comment by tackgyu** [comment]
@erdnussflips Thanks for the updates, but I did make it work by not passing swift version parameter at all, as:
`jazzy --podspec path/to/podspec --config path/to/config`
which is odd, because the project setting has never changed since the last try. I will 

I will leave this issue open and try your suggestion if I encounter the same issue.

**Comment by johnfairh** [comment]
Fixed in master via #1037.
